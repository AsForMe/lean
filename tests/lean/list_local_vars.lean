-- #exit
open interactive tactic.interactive

namespace tactic.interactive
open lean.parser

meta def tac (ls : parse lean.parser.list_include_var_names) : tactic unit :=
trace ls

open interactive.types
@[user_command]
meta def tac_cmd (_ : parse $ tk "stuff") : lean.parser unit :=
with_local_scope $
do ls ← lean.parser.list_include_variables,
   (n,t) ← brackets "(" ")" (prod.mk <$> (ident <* tk ":") <*> texpr),
   t ← tactic.to_expr t,
   v ← tactic.mk_local_def n t,
   lean.parser.add_local v,
   (n',t') ← brackets "(" ")" (prod.mk <$> (ident <* tk ":") <*> texpr),
   t' ← tactic.to_expr t',
   v ← tactic.mk_local_def n' t',
   lean.parser.add_local v,
   trace ls,
   brackets "(" ")" texpr,
   ↑(ls.mmap' $ trace ∘ to_string)

end tactic.interactive
-- stuff 1
variables (a b c : ℕ)
include a b
variables {α : Type}
section
stuff (β : Type) (γ : Type) (α × β × γ)
def x := β

#check x
section
parameter δ : Type

#print δ
#print β
end
#print γ

end

example : c ≤ 3 :=
begin
  (do v ← tactic.get_local `a,
      trace $ v.local_pp_name ),
  tac
end
