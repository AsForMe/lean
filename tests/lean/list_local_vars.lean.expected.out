[b, a]
b
a
list_local_vars.lean:34:9: error: unknown identifier 'β'
list_local_vars.lean:34:4: error: don't know how to synthesize placeholder
context:
a b : ℕ
⊢ Sort ?
x : ℕ → ℕ → ⁇
parameter δ : Type
list_local_vars.lean:42:0: error: unknown identifier β
list_local_vars.lean:45:0: error: unknown identifier γ
a
[b, a]
list_local_vars.lean:52:0: error: tactic failed, there are unsolved goals
state:
a b c : ℕ
⊢ c ≤ 3
